google/cloud/forseti/common/gcp_type/ke_cluster.py

```diff
@@ -89,22 +91,24 @@ def __init__(self, cluster_id, parent=None, full_name=None, locations=None,
         self._dict = None
 
     @classmethod
-    def from_json(cls, parent, json_string):
+    def from_json(cls, parent, json_string, cluster_id=None):
```

Comment:
Unfortunately  adding cluster_id here will not work too well for resource_util that calls from_json in a generic way: https://github.com/GoogleCloudPlatform/forseti-security/blob/dev/google/cloud/forseti/common/gcp_type/resource_util.py#L106

Can we deduce the cluster_id from the JSON response? The cluster.name being passed in as a kwarg must be originally have been created from the JSON response, so it should be possible.

Reply:
Updated to generate the key on the fly using hash(selfLink)